LocalyticsPlugin/Assets/Localytics/Localytics.cs

```diff
@@ -8,1073 +8,2163 @@
+				}
+
+				if (obj.LastName != null)
+				{
+					IntPtr method = AndroidJNIHelper.GetMethodID(builder.GetRawClass(), "setLastName", "(Ljava/lang/String;)Lcom.localytics.android.Customer$Builder;");
+					args[0] = obj.LastName;
+					AndroidJNI.CallObjectMethod(builder.GetRawObject(), method, AndroidJNIHelper.CreateJNIArgArray(args));
+				}
+
+				if (obj.FullName != null)
+				{
+					IntPtr method = AndroidJNIHelper.GetMethodID(builder.GetRawClass(), "setFullName", "(Ljava/lang/String;)Lcom.localytics.android.Customer$Builder;");
+					args[0] = obj.FullName;
+					AndroidJNI.CallObjectMethod(builder.GetRawObject(), method, AndroidJNIHelper.CreateJNIArgArray(args));
+				}
+
+				if (obj.EmailAddress != null)
+				{
+					IntPtr method = AndroidJNIHelper.GetMethodID(builder.GetRawClass(), "setEmailAddress", "(Ljava/lang/String;)Lcom.localytics.android.Customer$Builder;");
+					args[0] = obj.EmailAddress;
+					AndroidJNI.CallObjectMethod(builder.GetRawObject(), method, AndroidJNIHelper.CreateJNIArgArray(args));
+				}
+			}
+
+			return builder.Call<AndroidJavaObject> ("build", new object[]{});
+		}
+
+		private static List<CircularRegionInfo> ConvertGeofencesList(AndroidJavaObject listObject)
+		{
+			int size = listObject.Call<int>("size");
+
+			List<CircularRegionInfo> regionInfos = new List<CircularRegionInfo>();
+			for (int i = 0; i < size; i++)
+			{
+				AndroidJavaObject regionObj = listObject.Call<AndroidJavaObject>("get", i);
+				CircularRegionInfo info = ConvertCircularRegionInfoFromJava(regionObj);
+				regionInfos.Add(info);
+			}
+			return regionInfos;
+		}
+
+		private static AndroidJavaObject ConvertRegionInfoList(List<CircularRegionInfo> regions)
+		{
+			var list = new AndroidJavaObject("java.util.ArrayList");
+
+			if (regions == null || regions.Count == 0)
+			{
+				return list;
+			}
+
+			foreach (CircularRegionInfo info in regions)
+			{
+				list.Call<bool>("add", ConvertCircularRegionInfoToJava(info));
+			}
+			return list;
+		}
+
+		private static CircularRegionInfo ConvertCircularRegionInfoFromJava(AndroidJavaObject obj)
+		{
+			CircularRegionInfo info = new CircularRegionInfo();
+			info.UniqueId = obj.Call<string>("getUniqueId");
+			info.Latitude = obj.Call<double>("getLatitude");
+			info.Longitude = obj.Call<double>("getLongitude");
+			info.Radius = obj.Call<int>("getRadius");
+			info.Name = obj.Call<string>("getName");
+			info.Type = obj.Call<string>("getType");
+			info.Attributes = MapToDictionary(obj.Call<AndroidJavaObject>("getAttributes"));
+			return info;
+		}
+
+		private static AndroidJavaObject ConvertCircularRegionInfoToJava(CircularRegionInfo info)
+		{
+			var builder = new AndroidJavaObject("com.localytics.android.CircularRegion$Builder");
+			if (info != null)
+			{
+				var args = new object[1];
+				// Only care about uniqueId b/c triggerRegions will pull the populated CircularRegion
+				// object from the DB
+				if (in
```

Comment:
"unity"? I is confused by that... why is that the right way to get the location object?